Fireworks/Geometry/src/FWRecoGeometryESProducer.cc

```diff
@@ -134,15 +144,13 @@ std::unique_ptr<FWRecoGeometry> FWRecoGeometryESProducer::produce(const FWRecoGe
     addME0Geometry(*fwRecoGeometry);
   }
   if (m_calo) {
-    edm::ESHandle<CaloGeometry> caloGeomH;
-    record.getRecord<CaloGeometryRecord>().get(caloGeomH);
-    m_caloGeom = caloGeomH.product();
+    m_caloGeom = &record.get(m_caloGeomToken);
     addCaloGeometry(*fwRecoGeometry);
   }
 
   if (m_timing) {
-    record.getRecord<CaloGeometryRecord>().getRecord<IdealGeometryRecord>().get("FastTimeBarrel", m_ftlBarrelGeom);
-    record.getRecord<CaloGeometryRecord>().getRecord<IdealGeometryRecord>().get("SFBX", m_ftlEndcapGeom);
+    m_ftlBarrelGeom = &record.getRecord<CaloGeometryRecord>().get(m_ftlBarrelGeomToken);
```

Comment:
Event `m_ftlBarrelGeom = &record.get(m_ftlBarrelGeomToken);` should work since the GetToken tells the system which record to use.

Reply:
It doesn't because we did not make `DependentRecordImplementation::get()` to look beyond the first level of dependent-on records, see also https://github.com/cms-sw/cmssw/issues/26748#issuecomment-497102360